### ISSUE TITLE
Image garbage collection failed: non-existent label "docker-images"

### ISSUE THREAD

**Issue by cnaize** [kubernetes/kubernetes#9005]
Hello!
I know that in `kubernetes` where is garbage collection mechanism for `docker`.
I thought that it works from box, but I have `Image garbage collection failed: non-existent label "docker-images"` error.
How to fix it? How to cleanup `docker`'s `containers` and `images`?

**Comment by rjnagal** [comment]
@cnaize Can you add the output for 'docker info' and 'cat /proc/mounts'?

**Comment by cnaize** [comment]
@rjnagal 
docker info:
```none
Containers: 44
Images: 5
Storage Driver: devicemapper
 Pool Name: docker-253:1-70823815-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 347.5 MB
 Data Space Total: 107.4 GB
 Data Space Available: 13.03 GB
 Metadata Space Used: 2.654 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Udev Sync Supported: true
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Execution Driver: native-0.2
Kernel Version: 3.10.0-229.1.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 1
Total Memory: 489.6 MiB
Name: node1
ID: 3Y2W:PZQW:XXXB:4O44:P3M2:FWPJ:XWL7:IH6D:Q4SL:MVPA:YLFN:E72C
```
cat /proc/mounts:
```none
rootfs / rootfs rw 0 0
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
sysfs /sys sysfs rw,seclabel,nosuid,nodev,noexec,relatime 0 0
devtmpfs /dev devtmpfs rw,seclabel,nosuid,size=241024k,nr_inodes=60256,mode=755 0 0
securityfs /sys/kernel/security securityfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /dev/shm tmpfs rw,seclabel,nosuid,nodev 0 0
devpts /dev/pts devpts rw,seclabel,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 0 0
tmpfs /run tmpfs rw,seclabel,nosuid,nodev,mode=755 0 0
tmpfs /sys/fs/cgroup tmpfs rw,seclabel,nosuid,nodev,noexec,mode=755 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd 0 0
pstore /sys/fs/pstore pstore rw,nosuid,nodev,noexec,relatime 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpuacct,cpu 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
cgroup /sys/fs/cgroup/net_cls cgroup rw,nosuid,nodev,noexec,relatime,net_cls 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,nosuid,nodev,noexec,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/hugetlb cgroup rw,nosuid,nodev,noexec,relatime,hugetlb 0 0
configfs /sys/kernel/config configfs rw,relatime 0 0
/dev/mapper/centos-root / xfs rw,seclabel,relatime,attr2,inode64,noquota 0 0
selinuxfs /sys/fs/selinux selinuxfs rw,relatime 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs rw,relatime,fd=32,pgrp=1,timeout=300,minproto=5,maxproto=5,direct 0 0
hugetlbfs /dev/hugepages hugetlbfs rw,seclabel,relatime 0 0
mqueue /dev/mqueue mqueue rw,seclabel,relatime 0 0
debugfs /sys/kernel/debug debugfs rw,relatime 0 0
sunrpc /var/lib/nfs/rpc_pipefs rpc_pipefs rw,relatime 0 0
nfsd /proc/fs/nfsd nfsd rw,relatime 0 0
/dev/sda1 /boot xfs rw,seclabel,relatime,attr2,inode64,noquota 0 0
none /media/sf_sf_minebook vboxsf rw,nodev,relatime 0 0
gvfsd-fuse /run/user/1000/gvfs fuse.gvfsd-fuse rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0
fusectl /sys/fs/fuse/connections fusectl rw,relatime 0 0
none /home/node1/minebook vboxsf rw,nodev,relatime 0 0
none /home/node1/minebook vboxsf rw,nodev,relatime 0 0
none /home/node1/minebook vboxsf rw,nodev,relatime 0 0
```

**Comment by rjnagal** [comment]
Thanks for reporting @cnaize 

The labels don't show up because cadvisor doesn't look for filesystems other than btrfs and ext*.
I just added xfs to the list in https://github.com/google/cadvisor/pull/747

We'll pull it into kubelet soon. 

If you bring up a standalone cadvisor from head, you should notice "docker-images" label in storage endpoint with this change:
host:port/api/2.0/storage

**Comment by dalanlan** [comment]
Got the same pb. @rjnagal  
```
# docker info 

Containers: 45
Images: 68
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 158
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-32-generic
Operating System: Ubuntu 14.04.1 LTS
CPUs: 4
Total Memory: 7.798 GiB
Name: ubuntu
ID: SIUF:ZVAW:2IU3:2YMQ:PE6I:KSSJ:K523:HFBJ:2XLI:A774:ZSDF:25GO
WARNING: No swap limit support
```

And kubelet still with logs as following
```
E0723 07:30:30.712571       1 kubelet.go:682] Image garbage collection failed: non-existent label "docker-images"
```

Also no `docker-images` label
```
# curl http://127.0.0.1:4194/api/v2.0/storage
[{"device":"/dev/mapper/ubuntu--vg-root","mountpoint":"/rootfs","capacity":48197730304,"available":40347947008,"usage":5377871872,"labels":[]},{"device":"/dev/sda1","mountpoint":"/rootfs/boot","capacity":246755328,"available":196062208,"usage":37953536,"labels":[]}]
```

**Comment by mingfang** [comment]
+1

**Comment by rjnagal** [comment]
The fix for cadvisor was done a while back. I think we need to update cadvisor dependencies in kubelet.
I'll update the deps this week.

**Comment by yujuhong** [comment]
cadvisor dependencies have been updated.

**Comment by mingfang** [comment]
what was the fix for this?
I'm still seeing this problem in v1.2.4.

```
docker info
Containers: 39
 Running: 39
 Paused: 0
 Stopped: 0
Images: 217
Server Version: 1.11.0
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: null host bridge
Kernel Version: 4.2.0-35-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 31.3 GiB
Name: tristar
ID: 36CF:VRFA:ROLJ:YPVF:6L35:WA6M:33TY:UIHG:TLXM:QV6S:V3FI:UAV3
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

```
curl localhost:4194/api/v2.0/storage|jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   386  100   386    0     0   533k      0 --:--:-- --:--:-- --:--:--  376k
[
  {
    "labels": [],
    "usage": 411884720128,
    "available": 489727807488,
    "capacity": 949887705088,
    "mountpoint": "/rootfs",
    "device": "/dev/dm-0"
  },
  {
    "labels": [],
    "usage": 82666496,
    "available": 152313856,
    "capacity": 247772160,
    "mountpoint": "/rootfs/boot",
    "device": "/dev/sdb2"
  },
  {
    "labels": [],
    "usage": 75124736,
    "available": 934271729664,
    "capacity": 984373821440,
    "mountpoint": "/rootfs/media/hd2",
    "device": "/dev/sda1"
  }
]
```
